### ISSUE TITLE
System.Net.Sockets.Tests - IOControl_SIOCATMARK_Windows_Success failing on Windows.7.Amd64.Open-Release-x86

### ISSUE THREAD

**Issue by ahsonkhan** [dotnet/corefx#27717]
From https://github.com/dotnet/corefx/pull/27713
Windows.7.Amd64.Open-Release-x86
System.Net.Sockets.Tests.OSSupportTest/IOControl_SIOCATMARK_Windows_Success
https://mc.dot.net/#/user/dotnet-maestro-bot/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/79e54627d75ab93d5654d9803b57c9ea305cfb9f/workItem/System.Net.Sockets.Tests/analysis/xunit/System.Net.Sockets.Tests.OSSupportTest~2FIOControl_SIOCATMARK_Windows_Success
```text
Message :
Assert.Equal() Failure
Expected: 0
Actual:   1
Stack Trace :
   at System.Net.Sockets.Tests.OSSupportTest.IOControl_SIOCATMARK_Windows_Success() in D:\j\workspace\windows-TGrou---f8ac6754\src\System.Net.Sockets\tests\FunctionalTests\OSSupport.cs:line 166
```

**Comment by stephentoub** [comment]
cc: @rmkerr

**Comment by rmkerr** [comment]
Looking into this now. The failure occurs here, right after the data has been sent to the client:
https://github.com/dotnet/corefx/blob/ae595754d5977df993c7a9e96e26af9532a3977f/src/System.Net.Sockets/tests/FunctionalTests/OSSupport.cs#L165-L167
My best guess is that it's just a timing issue, since the test assumes that the data is received instantaneously after being sent. That isn't the case in reality, so the test may need a minor rewrite.

**Comment by rmkerr** [comment]
Fixed in #27838
